### PR TITLE
Updated KOIN dependencies

### PR DESCRIPTION
**Pull request by EngrTaofeek**
Updated KOIN dependencies to the recent ones more suitable for compose and fixed imports related errors. Also updated navigation-compose to be compatible with that version of KOIN.

### CODE REVIEW

**Review by Elforama — CHANGES_REQUESTED** [review]
Please undo start imports

File: app/build.gradle [hunk 1]
```diff
@@ -206,7 +206,14 @@ dependencies {
     implementation 'com.amazonaws:aws-android-sdk-core:2.16.8'
     implementation 'com.amazonaws:aws-android-sdk-s3:2.16.8'
 
-    implementation 'org.koin:koin-android-viewmodel:2.0.0-GA'
+    //koin dependencies
+    implementation "io.insert-koin:koin-core:$koin_version"
+    implementation "io.insert-koin:koin-android:$koin_version"
+    implementation "io.insert-koin:koin-android-compat:$koin_version"
```
Comment (Elforama) [inline app/build.gradle:212]: compat is for Java, so we shouldn't need it

File: app/build.gradle [hunk 2]
```diff
@@ -206,7 +206,14 @@ dependencies {
...
+    implementation "io.insert-koin:koin-androidx-workmanager:$koin_version"
```
Comment (Elforama) [inline app/build.gradle:213]: Unless there are issues with our current usage of Koin in our workmanager, I don't think we need this.

File: app/build.gradle [hunk 3]
```diff
@@ -206,7 +206,14 @@ dependencies {
...
+    implementation "io.insert-koin:koin-androidx-navigation:$koin_version"
```
Comment (Elforama) [inline app/build.gradle:214]: do you know what this navigation module adds? I can't find any documentation on it.

File: app/build.gradle [hunk 4]
```diff
@@ -206,7 +206,14 @@ dependencies {
     implementation 'com.amazonaws:aws-android-sdk-core:2.16.8'
     implementation 'com.amazonaws:aws-android-sdk-s3:2.16.8'
 
-    implementation 'org.koin:koin-android-viewmodel:2.0.0-GA'
+    //koin dependencies
+    implementation "io.insert-koin:koin-core:$koin_version"
+    implementation "io.insert-koin:koin-android:$koin_version"
```
Comment (Elforama) [inline app/build.gradle:211]: Do we need both of these? I'm curious if the android version contains core in it already

File: app/build.gradle (hunk 1 above)
Comment (EngrTaofeek) [reply app/build.gradle:212]: Okay I didn't know that.

File: app/build.gradle (hunk 4 above)
Comment (EngrTaofeek) [reply app/build.gradle:211]: Not both. I just realised Koin android alone is enough for the project.

**Comment by Elforama** [comment]
I think you might have to rebase. You have other people commits showing up.

**Comment by EngrTaofeek** [comment]
> I think you might have to rebase. You have other people commits showing up.

I rebased that's why it added the commits merged after I created the branch.

**Pull request merged**